app/protocols/theme-handler.js

```diff
@@ -0,0 +1,54 @@
+import path from 'path'
+import { fileURLToPath } from 'url'
+import fs from 'fs'
+import mime from 'mime-types'
+import { Readable } from 'stream'
+
+const __dirname = fileURLToPath(new URL('./', import.meta.url))
+
+export async function createThemeHandler () {
+  return {
+    handler: async function protocolHandler ({ url }, sendResponse) {
+      const parsedUrl = new URL(url)
+      let filePath
+
+      if (parsedUrl.hostname === 'theme') {
+        const fileName = parsedUrl.pathname.slice(1)
+        filePath = path.join(__dirname, '../pages/theme', fileName)
+        console.log('Attempting to serve:', filePath)
+
+        if (!fs.existsSync(filePath)) {
```

Comment:
Avoid using sync methods whenever possible. In this case we should use the existing `resolveFile` logic from the agregore handler. https://github.com/AgregoreWeb/agregore-browser/blob/master/app/protocols/browser-protocol.js#L154